app/elastic/es_search_runner.py

```diff
@@ -106,17 +107,23 @@ def get_es_search_response():
             cache_key,
             get_es_search_response,
             self.should_cache_search_response,
-            TIME_TO_LIVE,
+            get_cache_time_to_live(self.search_params),
         )
 
         self.total_results = cached_search_results["total_results"]
         self.es_search_results = cached_search_results["response"]
         self.execution_time = cached_search_results["execution_time"]
 
     def should_cache_search_response(self):
-        """Cache search response if execution time is higher than 400 ms"""
+        """Cache search response if execution time is higher than 400 ms
+        or if the query terms are a siren or a siret."""
         try:
-            if self.execution_time > MIN_EXECUTION_TIME:
+            query_terms = self.search_params.terms
+            if (
+                self.execution_time > MIN_EXECUTION_TIME
+                or is_siren(query_terms)
+                or is_siret(query_terms)
+            ):
```

Comment:
Another implem would be to have `should_cache_search_response` returning a number rather than a boolean
- `0` means no caching
- `> 0` means cache & TTL